### PR TITLE
feat: E2E tests iteration #2

### PR DESCRIPTION
**Pull request by laurentluce**
- Validate that the kardinal manager pods are running
- Validate access to the baseline and dev flows using the CLI gateway
- Validate the the dev pod terminates when the flow is deleted
- Validate that the kardinal manager pods are terminated after the manager is removed

### CODE REVIEW

File: .github/workflows/ci-e2e-tests.yml [hunk 1]
```diff
@@ -90,34 +90,74 @@ jobs:
...
+              kubectl get pods -n default -o custom-columns=NAMESPACE:metadata.namespace,POD:metadata.name,PodIP:status.podIP,READY-true:status.containerStatuses[*].ready
+              ((c++)) && ((c==12)) && exit 1
+              sleep 10
+          done
+          apps=$(kubectl get pods -n default -o custom-columns=:metadata.labels.app | tr " " "\n" | sort -g | tr "\n" " " | xargs)
+          echo ${apps}
+          if [ "${apps}" != "kardinal-manager trace-router trace-router-redis" ]; then exit 1; fi
+          
+          # Check for errors in the kardinal manager logs
+          if kubectl logs -n default -l dev.kardinal.app-id=kardinal-manager | grep "ERRO"
```
Comment (leoporoli) [inline .github/workflows/ci-e2e-tests.yml:106]: Wouldn't grep return 1 if it doesn't find the "ERRO" string?
And you can add the `-q` flag to set the quit mode to not print the output

File: .github/workflows/ci-e2e-tests.yml [hunk 2]
```diff
@@ -90,34 +90,74 @@ jobs:
...
+              sleep 10
+          done
+          flow_url=$(grep "http" kardinal.out | awk '{print $NF}')
+          status_code=$(curl -s -o /dev/null -w "%{http_code}" ${flow_url})
+          echo "${flow_url} returned status code ${status_code}"
+          if [ "${status_code}" != "200" ]; then exit 1; fi
+          kill ${cli_pid}
+
+          # Check for errors in the trace router logs
+          if kubectl logs -n default -l app=trace-router | grep "ERRO"
```
Comment (leoporoli) [inline .github/workflows/ci-e2e-tests.yml:155]: same comment here

File: .github/workflows/ci-e2e-tests.yml [hunk 3]
```diff
@@ -126,21 +166,54 @@ jobs:
...
+              sleep 10
+          done
+          flow_url=$(grep "http" kardinal.out | awk '{print $NF}')
+          status_code=$(curl -s -o /dev/null -w "%{http_code}" ${flow_url})
+          echo "${flow_url} returned status code ${status_code}"
+          if [ "${status_code}" != "200" ]; then exit 1; fi
+          kill ${cli_pid}
           
+          # Check for errors in the trace router logs
+          if kubectl logs -n default -l app=trace-router | grep "ERRO"
```
Comment (leoporoli) [inline .github/workflows/ci-e2e-tests.yml:195]: same here

**Review by leoporoli — APPROVED** [review]

File: .github/workflows/ci-e2e-tests.yml (hunk 1 above)
Comment (laurentluce) [reply .github/workflows/ci-e2e-tests.yml:106]: The if condition is fulfilled when grep exits with 0 (match found).  I will add the quiet flag.

**Pull request merged**